### DISCUSSION TITLE
How to link image to table row?

### DISCUSSION THREAD

**Discussion by oguruma1218** [supabase/supabase#14340] [Questions]
What's the best way to link an image, and multiple images, to a table row?

For example, I have a table of "Books" and each Book has a single "Cover_Image" and an "Image_Gallery" of multiple images.

How do I related the images/media to the Table row?

**Comment by GaryAustin1** [comment]
Not sure about best 
1. Store the paths, including an  array of paths for the image gallery in your product table. 
2. Use a image path table with fk link to product and path.
3. Use a folder with product id to filter all the images with storage api. 

For 1 and 2 you could store object id, but that involves a fetch to get it and then a fetch to to get the path from the objects table as the API only deals with paths.   

I would do 2 in your case.
